src/funding/funding.js

```diff
@@ -27,30 +27,30 @@ export function isFundingEligible(source : $Values<typeof FUNDING>, { layout, pl
         return false;
     }
 
-    if (layout && fundingConfig.layouts && fundingConfig.layouts.indexOf(layout) === -1) {
+    if (layout && fundingConfig.layouts && fundingConfig.layouts.indexOf(layout) === -1 && !enableFunding.some(funding => funding === fundingSource)) {
         return false;
     }
 
     if (fundingConfig.platforms && fundingConfig.platforms.indexOf(platform) === -1) {
         return false;
     }
 
-    if (fundingConfig.shippingChange === false && onShippingChange) {
+    if (fundingConfig.shippingChange === false && onShippingChange && !enableFunding.some(funding => funding === fundingSource)) {
```

Comment:
This should not be necessary -- even if merchant passes enable funding, shipping change will not be supported.

Reply:
Right, but we would like to show the buttons regardless even though apms will not support horizontal layout and shipping changes. This should be an integration error, since merchants called out they want this apm. We are planning to add some error message for these two use cases.

Reply:
If it's an integration error, why even show the buttons?